connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java

```diff
@@ -312,6 +313,9 @@ public void run() {
                         try {
                             readToLogEnd();
                             log.trace("Finished read to end log for topic {}", topic);
+                        } catch (TimeoutException e) {
+                            log.warn("Timeout while reading log to end {}", e.getMessage());
```

Comment:
How about putting the topic name in the message (like on line 315), denoting that this will be retried automatically, and maybe why this might happen.
```suggestion
                            log.warn("Timeout while reading log to end for topic '{}'. Retrying automatically. This may occur when brokers are unavailable or unreachable.", topic, e.getMessage());
```

Reply:
Good idea, though looks like this suggested change will exclude the message (doesn't have another `{}`); I'll fix that.